src/display-x11.cc

```diff
@@ -679,6 +423,314 @@ bool display_output_x11::main_loop_wait(double t) {
   return true;
 }
 
+#define EV_HANDLER(name)                                                     \
```

Comment:
I'm not a fan of adding new macros. They make it difficult to debug things, and they're not type safe. Probably better to just write out the whole function definition, or you could get fancy with templates (but I think in this case it's not worth the effort).

Reply:
Replaced with a template - it's more readable as it reduces the need for `#else`.

I used macros to simplify changing handler arguments (one source of truth), but I guess those won't be changing that often.